hyperswitch/HyperViewModel.swift

```diff
@@ -67,21 +67,32 @@ class HyperViewModel: ObservableObject {
     func preparePaymentMethodManagement() {
         Task {
             do {
-                let paymentMethodsJson = try await fetchData(from: "/payment_methods")
-                guard let publishableKey = paymentMethodsJson["publishableKey"] as? String,
-                      let paymentManagementClientSecret = paymentMethodsJson["clientSecret"] as? String else {
-                    throw NSError(domain: "API Error", code: 500, userInfo: [NSLocalizedDescriptionKey: "Missing required fields"])
-                }
+//                let paymentMethodsJson = try await fetchData(from: "/payment_methods")
+//                guard let publishableKey = paymentMethodsJson["publishableKey"] as? String,
+//                      let paymentManagementClientSecret = paymentMethodsJson["clientSecret"] as? String else {
+//                    throw NSError(domain: "API Error", code: 500, userInfo: [NSLocalizedDescriptionKey: "Missing required fields"])
+//                }
                 
                 let ephemeralKeyJson = try await fetchData(from: "/create-ephemeral-key")
                 guard let ephemeralKey = ephemeralKeyJson["ephemeralKey"] as? String else {
                     throw NSError(domain: "API Error", code: 500, userInfo: [NSLocalizedDescriptionKey: "Missing ephemeral key"])
                 }
                 
+                let json = try await fetchData(from: "/create-payment-intent")
+                guard let paymentIntentClientSecret = json["clientSecret"] as? String,
+                      let publishableKey = json["publishableKey"] as? String
+                else {
+                    throw NSError(domain: "API Error", code: 500, userInfo: [NSLocalizedDescriptionKey: "Missing required fields"])
+                }
+                
                 DispatchQueue.main.async {
                     self.status = .success
                     self.paymentSession = PaymentSession(publishableKey: publishableKey)
-                    self.paymentSession?.initPaymentManagementSession(ephemeralKey: ephemeralKey, paymentIntentClientSecret: paymentManagementClientSecret)
+
+                    // NOTE: since we are going with the "$0.0 payment-intent" approach, we don't need to pass the paymentManagementClientSecret, 
```

Comment:
remove comments here, this information can be provided in PR description

Reply:
removed 👍